apps/generator/test/integration.test.js

```diff
@@ -153,4 +154,62 @@ describe('Integration testing generateFromFile() to make sure the result of the
        Include log message test in the future to ensure that the log.debug for skipping overwrite is called
      */
   });
+
+  it('should not generate the conditionalFolder if the singleFile parameter is set true', async () => {
+    const outputDir = generateFolderName();
+    const generator = new Generator(reactTemplate, outputDir, {
+      forceWrite: true ,
+      templateParams: { version: 'v1', mode: 'production', singleFile: true }
+    });
+    await generator.generateFromFile(dummySpecPath);
+    const conditionalFolderPath = path.join(outputDir, 'conditionalFolder');
+    const exists = fs.existsSync(conditionalFolderPath);
+    expect(exists).toBe(false);
+  });
+  
+  it('should generate the conditionalFolder if the singleFile parameter is set false by default', async () => {
+    const outputDir = generateFolderName();
+    const generator = new Generator(reactTemplate, outputDir, {
+      forceWrite: true ,
+      templateParams: { version: 'v1', mode: 'production' }
+    });
+    await generator.generateFromFile(dummySpecPath);
+    const conditionalFolderPath = path.join(outputDir, 'conditionalFolder');
+    const exists = fs.existsSync(conditionalFolderPath);
+    expect(exists).toBe(true);
+  });
+
+  it('should not generate the conditionalFolder in one by one process in a single folder, if the singleFile parameter is set true', async () => {
+    const outputDir = generateFolderName();
+    for (let i = 1; i <= 3; i++) {
```

Comment:
Why do we need these 2 test cases? They are not really testing race conditions. A race condition took place because of hooks in html-template, that were removing folders. In here you basically run 3 generations one after another

Reply:
Added one test case for this thanks @derberg